src/main/java/school/lemon/changerequest/java/generics/generator/NumberGenerator.java

```diff
@@ -0,0 +1,13 @@
+package school.lemon.changerequest.java.generics.generator;
+
+import school.lemon.changerequest.java.generics.container.GenericContainer;
+
+import java.util.Random;
+
+public abstract class NumberGenerator {
```

Comment:
The purpose here was to define a generic type on the class level.

Reply:
What sence to do class generic if it doesn`t contain generic fields? Just for clarity?

Reply:
It gives you more context about the class usage.
E.g.:
```java
NumberGenerator<Integer> generator = new IntegerGenerator();
```

With such declaration you can pass a more generic type to some function as an argument, of use it as a class field. It doesn't restrict you to use the implementation.

What numbers would be generated by the following code?
```java
void doGenerate(NumberGenerator generator){
TYPE numbers = generator.generateNumbers();
}
```